cmd/main.go

```diff
@@ -1,124 +1,86 @@
 package main
 
 import (
-	"encoding/json"
 	"fmt"
+	"io"
 	"log"
 	"net/http"
-	"net/url"
 	"os"
-	"strconv"
-	"time"
 
 	"go.uber.org/automaxprocs/maxprocs"
 
-	"github.com/chrisbrown1111/price-service/pkg/jwt"
+	// "github.com/chrisbrown1111/price-service/pkg/jwt"
+	"github.com/chrisbrown1111/price-service/pkg/downstream"
 	"github.com/chrisbrown1111/price-service/pkg/postgres"
 )
 
-// Define a struct that matches the JSON structure
-type DiscountResponse struct {
-	Discount           float32  `json:"discount"`
-	Item               string   `json:"item"`
-	Quantity           int      `json:"quantity"`
-	Applicable_in_eu   bool     `json:"applicable_in_eu"`
-	Shipping_cost      float32  `json:"shipping_cost"`
-	Shipping_time_days int      `json:"shipping_time_days"`
-	Related_items      []string `json:"related_items"`
-}
-
-func priceHandler(w http.ResponseWriter, r *http.Request) {
-
-	jwtToken := r.URL.Query().Get("jwt")
-
-	secretKey, err := jwt.GetJwtKey()
-	if err != nil {
-		log.Fatalf("priceHandler getJwtKey err: %v", err)
-	}
-
-	claims, err := jwt.ParseJWT(jwtToken, secretKey)
-	if err != nil {
-		log.Printf("Error parsing JWT: %v", err)
-		w.WriteHeader(http.StatusBadRequest)
-		return
-	}
-
-	// Fetch the price from Postgres
-	price, err := postgres.FetchPrice(claims.Item)
-	if err != nil {
-		log.Printf("fetchPrice %v", err)
-		return
+func newPriceHandler() (http.HandlerFunc, error) {
+	remoteServer := ""
+	if s := os.Getenv("REMOTE_SERVER"); s == "" {
+		return nil, fmt.Errorf("env var REMOTE_SERVER not found")
+	} else {
+		remoteServer = s
 	}
+	dsClient := downstream.New(remoteServer)
```

Comment:
This moves the client creation out as a one-time operation. We were initialising the http client on every request before, which we would never do...